core/lib/tasks/states.rake

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+def get_carmen_states(iso)
+  carmen_country = Carmen::Country.coded(iso)
+  return false unless carmen_country&.subregions?
+
+  if %w(IT).include? iso
+    return carmen_country.subregions.flat_map(&:subregions)
+  else
+    return carmen_country.subregions
+  end
+end
+
+def user_agrees(destroy_states, create_states, country)
+  puts "\e[31mStates for #{country.name} are incorrect. If you proceed, the following changes will happen:\e[0m"
+  puts "\e[31mThese states will be destroyed:\e[0m #{destroy_states}"
+  puts "\e[31mThese states will be created:\e[0m #{create_states}"
+  puts "\e[31mDo you want to proceed? [y/n]\e[0m "
+  ["y", "yes"].include? STDIN.gets.strip.downcase
+end
+
+def delete_states(country, carmen_names)
+  country.states.where.not(name: carmen_names).destroy_all
```

Comment:
I'm hesitant doing this since the addresses stored for old orders will lose the database reference to these states. I suppose we can't soft delete states, right?

Reply:
We can, AFAIK we'd just have to add a "discarded_at" index, as per the discard readme: https://github.com/jhawthorn/discard

I think that might be a good option, but if that's too much we can also just warn the user that the state was unable to be deleted, and let them handle it however they want.

Reply:
Honestly, I would not provide this rake task to users. It's too dangerous. I think that leaving this code in the PR's description as one of the possible migration paths (making clear that states will be destroyed and addresses that use those states will lose references) is enough for people that want to migrate their data after this change. I don't expect to be a lot since if they sell in those regions they probably already did the change. 

For the others that start a new Solidus store this is not an issue since they'll have their code updated

Reply:
Okay, I've moved it to a gist and added a link to the PR description.